project-base/storefront/components/Blocks/Product/LastVisitedProducts/LastVisitedProducts.tsx

```diff
@@ -0,0 +1,49 @@
+import { ProductsSlider } from 'components/Blocks/Product/ProductsSlider';
+import { SkeletonModuleLastVisitedProducts } from 'components/Blocks/Skeleton/SkeletonModuleLastVisitedProducts';
+import { Webline } from 'components/Layout/Webline/Webline';
+import { useProductsByCatnumsApi } from 'graphql/generated';
+import { GtmProductListNameType } from 'gtm/types/enums';
+import { useLastVisited } from 'helpers/products/lastVisitedProducts';
+import useTranslation from 'next-translate/useTranslation';
+
+type LastVisitedProductsProps = {
+    removeCatalogNumber?: string;
+};
+
+export const LastVisitedProducts: FC<LastVisitedProductsProps> = ({ removeCatalogNumber }) => {
+    const { getLastVisitedProductCatalogNumbers } = useLastVisited();
+    let lastVisitedProductsFromCookies = getLastVisitedProductCatalogNumbers();
+
+    if (removeCatalogNumber && !!lastVisitedProductsFromCookies.length) {
+        lastVisitedProductsFromCookies = lastVisitedProductsFromCookies.filter(
+            (item: string) => item !== removeCatalogNumber,
+        );
+    }
+
+    const [{ data: result, fetching }] = useProductsByCatnumsApi({
+        variables: { catnums: lastVisitedProductsFromCookies },
+    });
```

Comment:
I see the API is called even though we don't have any lastVisitedProducts. Let's fix this.

Reply:
Added pause param.